presto-main/src/main/java/com/facebook/presto/cost/PlanCostEstimate.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.cost;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.lang.Double.isNaN;
+
+public final class PlanCostEstimate
+{
+    public static PlanCostEstimate zero()
+    {
+        return new PlanCostEstimate(0, 0, 0, 0);
+    }
+
+    private final double cpuCost;
+    private final double maxMemory;
```

Comment:
is this peak or cumulative memory usage?

Reply:
what is cumulative memory usage?

Reply:
Very interesting metric! We do not calculate this today, but it would be valuable "v2" addition. Especially for clusters with some (or high) concurrency.

This is estimated maximal amount of memory consumed when executing a plan. I thought `maxMemory` conveys the intent, but maybe you have some better name?